cleanlab/cleanlab-studio
# Feature: tlm api fixes and concurrency

## Testing responsibilities:
### Hui Wen
please review the updates in `cleanlab_studio/studio/trustworthy_language_model.py` and the corresponding test files (`tests/tlm/test_get_trustworthiness_score.py` and `tests/tlm/test_prompt.py`
specifically, ensure API functionality is maintained (can use tests to aid with that validation)

### Luke
Please review the addition of the AI/MD concurrency controller (`cleanlab_studio/internal/tlm/concurrency.py`) and corresponding tests (`tests/tlm/test_concurrency.py`)

## Description

- Unifies as much concurrency code as possible
- Adds basic e2e tests
- Adds [AI/MD concurrency control](https://witestlab.poly.edu/respond/sites/genitutorial/module/tcp-original-concepts)

Testing instructions:
- Run tests (ensure you set your desired `CLEANLAB_API_BASE_URL` first and run `cleanlab login`)
- Test manually
